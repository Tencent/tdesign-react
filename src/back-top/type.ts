/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode } from '../common';

export interface TdBackTopProps {
  /**
   * 是否绝对定位固定到屏幕右下方
   * @default true
   */
  fixed?: boolean;
  /**
   * 图标
   * @default 'backtop'
   */
  icon?: TNode;
  /**
   * 定位滚动到指定对象
   * @default () => window
   */
  target?: () => HTMLElement;
  /**
   * 文案
   * @default ''
   */
  text?: string;
  /**
   * 预设的样式类型
   * @default round
   */
  theme?: 'default' | 'primary' | 'dark';

  /**
   *  组件的形状
   */
  shape?: 'square' | 'circle';

  /**
   * 组件的大小
   */
  size?: 'small' | 'medium';
}
