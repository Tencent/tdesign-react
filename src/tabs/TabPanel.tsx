import React from 'react';
import classNames from 'classnames';
import { TdTabPanelProps } from './type';
import { StyledProps } from '../common';
import { useTabClass } from './useTabClass';
import { tabPanelDefaultProps } from './defaultProps';

export interface TabPanelProps extends TdTabPanelProps, StyledProps {
  children?: React.ReactNode;
}

const TabPanel: React.FC<TabPanelProps> = (props) => {
  const { tdTabPanelClassPrefix } = useTabClass();

  const { className, style } = props;

  return (
    <div className={classNames(tdTabPanelClassPrefix, className)} style={style}>
      {props.children}
    </div>
  );
};

TabPanel.displayName = 'TabPanel';
TabPanel.defaultProps = tabPanelDefaultProps;

export default TabPanel;
