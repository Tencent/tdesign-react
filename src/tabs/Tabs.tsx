import React, { useEffect, useMemo, useState } from 'react';
import classNames from 'classnames';
import { TabValue, TdTabsProps } from './type';
import forwardRefWithStatics from '../_util/forwardRefWithStatics';
import TabNav from './TabNav';
import { useTabClass } from './useTabClass';
import TabPanel from './TabPanel';
import { StyledProps } from '../common';
import { tabsDefaultProps } from './defaultProps';

export interface TabsProps extends TdTabsProps, StyledProps {
  children?: React.ReactNode;
}

const Tabs = forwardRefWithStatics(
  (props: TabsProps, ref) => {
    const { children, list, placement, onRemove, value: tabValue, onChange, className, style } = props;
    let { defaultValue } = props;

    // 样式工具引入
    const { tdTabsClassPrefix, tdTabsClassGenerator, tdClassGenerator } = useTabClass();

    const memoChildren = useMemo(() => {
      if (!list || list.length === 0) {
        return children;
      }
      return list.map((panelProps) => <TabPanel key={panelProps.value} {...panelProps} />);
    }, [children, list]);

    const itemList = React.Children.map(memoChildren, (child: any) => {
      if (child && child.type === TabPanel) {
        return child.props;
      }
      return null;
    });

    // 当未设置默认值时，默认选中第一个。
    if (defaultValue === undefined && Array.isArray(itemList) && itemList.length !== 0) {
      defaultValue = itemList[0].value;
    }

    const [value, setValue] = useState<TabValue>(defaultValue);

    useEffect(() => {
      tabValue !== undefined && setValue(tabValue);
    }, [tabValue]);

    const handleChange = (v) => {
      if (tabValue === undefined) {
        setValue(v);
      }
      onChange?.(v);
    };

    const handleClickTab = (v) => {
      if (tabValue === undefined) {
        setValue(v);
      }
    };

    const renderHeader = () => (
      <div className={classNames(tdTabsClassGenerator('header'), tdClassGenerator(`is-${placement}`))}>
        <TabNav
          {...props}
          activeValue={value}
          onRemove={onRemove}
          itemList={itemList}
          tabClick={handleClickTab}
          onChange={handleChange}
        />
      </div>
    );

    return (
      <div ref={ref} className={classNames(tdTabsClassPrefix, className)} style={style}>
        {placement !== 'bottom' ? renderHeader() : null}
        <div className={classNames(tdTabsClassGenerator('content'), tdClassGenerator(`is-${placement}`))}>
          {React.Children.map(memoChildren, (child: any) => {
            if (child && child.type === TabPanel) {
              if (child.props.value === value) {
                return child;
              }
              if (child.props.destroyOnHide === false) {
                return <TabPanel style={{ display: 'none' }}>{child.props.children}</TabPanel>;
              }
            }
            return null;
          })}
        </div>
        {placement === 'bottom' ? renderHeader() : null}
      </div>
    );
  },
  { TabPanel },
);

Tabs.displayName = 'Tabs';
Tabs.defaultProps = tabsDefaultProps;

export default Tabs;
