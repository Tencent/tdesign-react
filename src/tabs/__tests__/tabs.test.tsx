import React from 'react';
import { render, testExamples, waitFor, fireEvent } from '@test/utils';
import TabPanel from '../TabPanel';
import Tabs from '../Tabs';
import TabNav from '../TabNav';
import { TdTabsProps } from '../type';

testExamples(__dirname);

describe('Tabs 组件测试', () => {
  test('render Tab bar', async () => {
    const testId = 'tab bar test id';
    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} data-testid={testId} size={'medium'}>
          <TabPanel value={'a'} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));

    expect(() => tabInstance.querySelector('.t-tabs__nav')).not.toBe(null);
  });

  test('render card theme', async () => {
    const testId = 'tab card theme test id';
    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'}>
          <TabPanel value={'a'} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));

    expect(() => tabInstance.querySelector('.t-tabs__nav--card')).not.toBe(null);
  });

  test('different position', async () => {
    const testId = 'tab position test id';
    const positions: TdTabsProps['placement'][] = ['top', 'bottom', 'left', 'right'];

    const { getByTestId } = render(
      <div data-testid={testId}>
        {positions.map((position, index: number) => (
          <Tabs placement={position} size={'medium'} key={index}>
            <TabPanel value={'a'} label={'a'}>
              <div>a</div>
            </TabPanel>
            <TabPanel value={'b'} label={'b'}>
              <div>b</div>
            </TabPanel>
          </Tabs>
        ))}
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));

    positions.forEach((position) => {
      expect(() => tabInstance.querySelector(`.t-is-${position}`)).not.toBe(null);

      if (['left', 'right'].includes(position)) {
        expect(() => tabInstance.querySelector(`.t-is-vertical`)).not.toBe(null);
      }
    });
  });

  test('remove tab', async () => {
    const testId = 'remove tab test id';

    const removeFn = jest.fn();

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'} onRemove={removeFn}>
          <TabPanel value={'a'} removable={true} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} removable={true} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));
    fireEvent.click(tabInstance.querySelector('.remove-btn'));
    expect(removeFn).toBeCalledTimes(1);
  });

  test('remove disabled tab', async () => {
    const testId = 'remove disabled tab test id';

    const removeFn = jest.fn();

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'} onRemove={removeFn} disabled>
          <TabPanel value={'a'} removable={true} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} removable={true} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));
    fireEvent.click(tabInstance.querySelector('.remove-btn'));
    expect(removeFn).toBeCalledTimes(0);
  });

  test('click tab item', async () => {
    const testId = 'tab item test id';

    const clickFn = jest.fn();

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'} value="b" onChange={clickFn}>
          <TabPanel value={'a'} removable={true} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} removable={true} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));
    fireEvent.click(tabInstance.querySelector('.t-tabs__nav-item'));
    expect(clickFn).toBeCalledTimes(1);
  });

  test('add tab item', async () => {
    const testId = 'add tab item test id';

    const addFn = jest.fn();

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'} value="b" onAdd={addFn} addable>
          <TabPanel value={'a'} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));
    fireEvent.click(tabInstance.querySelector('.t-tabs__add-btn'));
    expect(addFn).toBeCalledTimes(1);
  });

  test('no tab item', async () => {
    const testId = 'no tab item test id';

    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'} value="b">
          <div></div>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));
    expect(tabInstance.querySelector('.t-tabs__nav-item')).toBe(null);
  });

  test('test TabNav event', async () => {
    const testId = 'no tab item test id';

    const { getByTestId } = render(
      <div data-testid={testId}>
        <TabNav
          defaultValue={0}
          activeValue={0}
          itemList={[
            {
              value: 0,
            },
          ]}
          tabClick={undefined}
        ></TabNav>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));
    expect(tabInstance.querySelector('.t-tabs__nav-item-wrapper')).not.toBe(null);
  });
});
