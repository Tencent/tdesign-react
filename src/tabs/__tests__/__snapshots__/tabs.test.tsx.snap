// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ban.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <div
          class="t-tabs__operations t-tabs__operations--left"
        />
        <div
          class="t-tabs__operations t-tabs__operations--right"
        />
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
                style="transform: translateX(0px);"
              />
              <div
                class="t-tabs__bar t-is-top"
              />
              <div
                class="t-tabs__nav-item t-size-m t-is-top t-is-disabled"
              >
                <div
                  class="t-tabs__nav-item-wrapper"
                  style="position: relative;"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡一
                  </span>
                </div>
              </div>
              <div
                class="t-tabs__nav-item t-size-m t-is-active t-is-top"
              >
                <div
                  class="t-tabs__nav-item-wrapper"
                  style="position: relative;"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡二
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content t-is-top"
    >
      <div
        class="t-tab-panel"
      >
        <div
          style="margin: 20px;"
        >
          这是一个Tabs
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <div
          class="t-tabs__operations t-tabs__operations--left"
        />
        <div
          class="t-tabs__operations t-tabs__operations--right"
        />
        <div
          class="t-tabs__nav-container t-is-top"
        >
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
                style="transform: translateX(0px);"
              />
              <div
                class="t-tabs__bar t-is-top"
              />
              <div
                class="t-tabs__nav-item t-size-m t-is-active t-is-top"
              >
                <div
                  class="t-tabs__nav-item-wrapper"
                  style="position: relative;"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡1
                  </span>
                </div>
              </div>
              <div
                class="t-tabs__nav-item t-size-m t-is-top"
              >
                <div
                  class="t-tabs__nav-item-wrapper"
                  style="position: relative;"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡2
                  </span>
                </div>
              </div>
              <div
                class="t-tabs__nav-item t-size-m t-is-top"
              >
                <div
                  class="t-tabs__nav-item-wrapper"
                  style="position: relative;"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content t-is-top"
    >
      <div
        class="t-tab-panel"
      >
        <div
          class="tabs-content"
          style="margin: 20px;"
        >
          选项卡1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`combination.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <div
          class="t-tabs__operations t-tabs__operations--left"
        />
        <div
          class="t-tabs__operations t-tabs__operations--right"
        >
          <div
            class="t-tabs__add-btn t-tabs__btn t-size-m"
          >
            <svg
              class="t-icon t-icon-add"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-tabs__nav-container t-tabs__nav--card t-is-top t-is-addable"
        >
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
              <div
                class="t-tabs__nav-item t-tabs__nav--card t-size-m t-is-active t-is-top"
              >
                <span
                  class="t-tabs__nav-item-text-wrapper"
                >
                  选项卡1
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content t-is-top"
    >
      <div
        class="t-tab-panel"
      >
        <div
          class="tabs-content"
          style="margin: 20px;"
        >
          选项卡1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`custom.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <div
          class="t-tabs__operations t-tabs__operations--left"
        />
        <div
          class="t-tabs__operations t-tabs__operations--right"
        >
          <div
            class="t-tabs__add-btn t-tabs__btn t-size-m"
          >
            <svg
              class="t-icon t-icon-add"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-tabs__nav-container t-tabs__nav--card t-is-top t-is-addable"
        >
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
              <div
                class="t-tabs__nav-item t-tabs__nav--card t-size-m t-is-active t-is-top"
              >
                <span
                  class="t-tabs__nav-item-text-wrapper"
                >
                  选项卡1
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content t-is-top"
    >
      <div
        class="t-tab-panel"
      >
        <div
          class="tabs-content"
          style="margin: 20px;"
        >
          选项卡1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-tabs"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-outline"
      style="position: relative;"
      type="button"
    >
      <span
        class="t-button__text"
      >
        切换到卡片型
      </span>
    </button>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__operations t-tabs__operations--left"
          />
          <div
            class="t-tabs__operations t-tabs__operations--right"
          />
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__bar t-is-top"
                />
                <div
                  class="t-tabs__nav-item t-size-m t-is-active t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        <svg
                          class="t-icon t-icon-discount"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M7.84 5.01A2 2 0 115 7.84 2 2 0 017.84 5zm-.7.7a1 1 0 10-1.42 1.42 1 1 0 001.41-1.41z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M2 7.95V2h5.95a1 1 0 01.7.3l6.14 6.13a1 1 0 010 1.41L9.84 14.8a1 1 0 01-1.41 0L2.29 8.66A1 1 0 012 7.95zM7.95 3l6.13 6.13-4.95 4.95L3 7.95V3h4.95z"
                            fill="currentColor"
                            fill-opacity="0.9"
                            fill-rule="evenodd"
                          />
                        </svg>
                        选项卡
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        <svg
                          class="t-icon t-icon-discount"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M7.84 5.01A2 2 0 115 7.84 2 2 0 017.84 5zm-.7.7a1 1 0 10-1.42 1.42 1 1 0 001.41-1.41z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M2 7.95V2h5.95a1 1 0 01.7.3l6.14 6.13a1 1 0 010 1.41L9.84 14.8a1 1 0 01-1.41 0L2.29 8.66A1 1 0 012 7.95zM7.95 3l6.13 6.13-4.95 4.95L3 7.95V3h4.95z"
                            fill="currentColor"
                            fill-opacity="0.9"
                            fill-rule="evenodd"
                          />
                        </svg>
                        选项卡
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        <svg
                          class="t-icon t-icon-discount"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M7.84 5.01A2 2 0 115 7.84 2 2 0 017.84 5zm-.7.7a1 1 0 10-1.42 1.42 1 1 0 001.41-1.41z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M2 7.95V2h5.95a1 1 0 01.7.3l6.14 6.13a1 1 0 010 1.41L9.84 14.8a1 1 0 01-1.41 0L2.29 8.66A1 1 0 012 7.95zM7.95 3l6.13 6.13-4.95 4.95L3 7.95V3h4.95z"
                            fill="currentColor"
                            fill-opacity="0.9"
                            fill-rule="evenodd"
                          />
                        </svg>
                        选项卡
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content t-is-top"
      >
        <div
          class="t-tab-panel"
        >
          <div
            class="tabs-content"
            style="margin: 20px;"
          >
            选项卡1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`operation.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-tabs"
  >
    <div
      class="t-tabs__header t-is-top"
    >
      <div
        class="t-tabs__nav"
      >
        <div
          class="t-tabs__operations t-tabs__operations--left"
        />
        <div
          class="t-tabs__operations t-tabs__operations--right"
        >
          <div
            class="t-tabs__add-btn t-tabs__btn t-size-m"
          >
            <svg
              class="t-icon t-icon-add"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-tabs__nav-container t-tabs__nav--card t-is-top t-is-addable"
        >
          <div
            class="t-tabs__nav-scroll"
          >
            <div
              class="t-tabs__nav-wrap"
            >
              <div
                class="t-tabs__bar t-is-top"
              />
              <div
                class="t-tabs__nav-item t-tabs__nav--card t-size-m t-is-active t-is-top"
              >
                <span
                  class="t-tabs__nav-item-text-wrapper"
                >
                  选项卡1
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs__content t-is-top"
    >
      <div
        class="t-tab-panel"
      >
        <div
          style="margin: 20px;"
        >
          选项卡1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`position.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="tdesign-demo-block-row"
    >
      <button
        class="t-button t-button--theme-default t-button--variant-outline"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          top
        </span>
      </button>
      <button
        class="t-button t-button--theme-default t-button--variant-outline"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          bottom
        </span>
      </button>
      <button
        class="t-button t-button--theme-default t-button--variant-outline"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          left
        </span>
      </button>
      <button
        class="t-button t-button--theme-default t-button--variant-outline"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          right
        </span>
      </button>
    </div>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__operations t-tabs__operations--left"
          />
          <div
            class="t-tabs__operations t-tabs__operations--right"
          />
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__bar t-is-top"
                />
                <div
                  class="t-tabs__nav-item t-size-m t-is-active t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      选项卡1
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        选项卡2
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      选项卡3
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      选项卡4
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        选项卡5
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        选项卡6
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        选项卡7
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content t-is-top"
      >
        <div
          class="t-tab-panel"
        >
          <div
            class="tabs-content"
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`size.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="tdesign-demo-block-row"
    >
      <button
        class="t-button t-button--theme-default t-button--variant-outline"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          medium
        </span>
      </button>
      <button
        class="t-button t-button--theme-default t-button--variant-outline"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          large
        </span>
      </button>
    </div>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__operations t-tabs__operations--left"
          />
          <div
            class="t-tabs__operations t-tabs__operations--right"
          />
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__bar t-is-top"
                />
                <div
                  class="t-tabs__nav-item t-size-m t-is-active t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      选项卡一
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      选项卡二
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content t-is-top"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__operations t-tabs__operations--left"
          />
          <div
            class="t-tabs__operations t-tabs__operations--right"
          />
          <div
            class="t-tabs__nav-container t-tabs__nav--card t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
              >
                <div
                  class="t-tabs__bar t-is-top"
                />
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-size-m t-is-active t-is-top"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡一
                  </span>
                </div>
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-size-m t-is-top"
                >
                  <span
                    class="t-tabs__nav-item-text-wrapper"
                  >
                    选项卡二
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content t-is-top"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`theme.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <div
      class="tdesign-demo-block-row"
    >
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          default
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          card
        </span>
      </button>
    </div>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__operations t-tabs__operations--left"
          />
          <div
            class="t-tabs__operations t-tabs__operations--right"
          />
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__bar t-is-top"
                />
                <div
                  class="t-tabs__nav-item t-size-m t-is-active t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      1
                    </span>
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m t-is-top"
                >
                  <div
                    class="t-tabs__nav-item-wrapper"
                    style="position: relative;"
                  >
                    <span
                      class="t-tabs__nav-item-text-wrapper"
                    >
                      <div>
                        2
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content t-is-top"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
